### PR TITLE
Add Custom ULX moderation commands

### PR DESCRIPTION
**Pull request by wrefgtzweve**

### CODE REVIEW

File: lua/ulx/modules/sh/cfc_checklua.lua [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+if SERVER then
+    util.AddNetworkString( "CFC_ULX_StatCheckCL" )
+    util.AddNetworkString( "CFC_ULX_StatCheckSV" )
+end
+
+local awaitingResponse = {}
+
+net.Receive( "CFC_ULX_StatCheckSV", function( _, ply )
+    if not ply.waitingOnStatCheck then return end
+    local convar = net.ReadBool()
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_checklua.lua:15]: could use a better name
allowLuaConvar, isCSLuaAllowed, or something

File: lua/ulx/modules/sh/cfc_checklua.lua [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        ply.waitingOnStatCheck = true
+        awaitingResponse[ply] = callingPlayer
+        net.Start( "CFC_ULX_StatCheckCL" )
+        net.Send( ply )
+    end
+end
+
+if CLIENT then
+    net.Receive( "CFC_ULX_StatCheckCL", function()
+        local convarBool = GetConVar( "sv_allowcslua" ):GetBool()
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_checklua.lua:33]: also not a great variable name

File: lua/ulx/modules/sh/cfc_speed.lua [hunk 3]
```diff
@@ -0,0 +1,22 @@
+CFCUlxCommands.speed = CFCUlxCommands.speed or {}
+local cmd = CFCUlxCommands.speed
+
+CATEGORY_NAME = "Fun"
+                    
+function cmd.speedPlayers( callingPlayer, targetPlayers, amount )
+    for _, ply in pairs( targetPlayers ) do
+        local inPvp = hook.Run( "CFC_Ulxcommands_SpeedPvpCheck", ply )
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_speed.lua:8]: if the point of using hooks like this is to decouple the ulx command addon from the pvp addon a more generic hook name would be better e.g. `CFC_ULXCommands_CanApplySpeedModifier`

File: lua/ulx/modules/sh/cfc_checklua.lua [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    ply.waitingOnStatCheck = false
+    ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's sv_allowcslua value is " .. tostring( convar ), ply )
+
+    awaitingResponse[ply] = nil
+end )
+
+function cmd.checkluaPlayers( callingPlayer, targetPlayers )
+    for _, ply in pairs( targetPlayers ) do
+        ply.waitingOnStatCheck = true
+        awaitingResponse[ply] = callingPlayer
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_checklua.lua:25]: same thing as the previous command, is having both of these necessary?

**Review by plally — DISMISSED** [review]

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    local function getFriendStatus( ply )
+        if ply == LocalPlayer() then return end
+
+        local friendStatus = ply:GetFriendStatus()
+        if friendStatus == "none" then return end
+
+        return friendStatus
+    end
+
+    net.Receive( "CFC_ULX_CheckfriendsSend", function()
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:54]:
```suggestion
    net.Receive( "CFC_ULX_CheckFriendsSend", function()
```

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        friendTable = {}
+        local onlinePlayers = player.GetHumans()
+        for _, ply in pairs( onlinePlayers ) do
+            friendTable[ply] = getFriendStatus( ply )
+        end
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:59]: You're overwriting `ply` here

```suggestion
        for _, onlinePly in pairs( onlinePlayers ) do
            if onlinePly ~= ply then
                friendTable[onlinePly] = getFriendStatus( ply )
            end
        end
```

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+        net.Start( "CFC_ULX_CheckfriendsRecieve" )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:61]:
```suggestion
        net.Start( "CFC_ULX_CheckFriendsReceive" )
```

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+CATEGORY_NAME = "Utility"
+
+if SERVER then
+    util.AddNetworkString( "CFC_ULX_CheckfriendsSend" )
+    util.AddNetworkString( "CFC_ULX_CheckfriendsRecieve" )
+end
+
+local awaitingResponse = {}
+
+net.Receive( "CFC_ULX_CheckfriendsRecieve", function( _, ply )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:13]:
```suggestion
net.Receive( "CFC_ULX_CheckFriendsReceive", function( _, ply )
```

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 9]
```diff
@@ -0,0 +1,70 @@
+CFCUlxCommands.checkfriends = CFCUlxCommands.checkfriends or {}
+local cmd = CFCUlxCommands.checkfriends
+
+CATEGORY_NAME = "Utility"
+
+if SERVER then
+    util.AddNetworkString( "CFC_ULX_CheckfriendsSend" )
+    util.AddNetworkString( "CFC_ULX_CheckfriendsRecieve" )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:8]:
```suggestion
    util.AddNetworkString( "CFC_ULX_CheckFriendsSend" )
    util.AddNetworkString( "CFC_ULX_CheckFriendsReceive" )
```

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 10]
```diff
@@ -0,0 +1,70 @@
+CFCUlxCommands.checkfriends = CFCUlxCommands.checkfriends or {}
+local cmd = CFCUlxCommands.checkfriends
+
+CATEGORY_NAME = "Utility"
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:4]:
```suggestion
local CATEGORY_NAME = "Utility"
```

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+        return
+    end
+    
+    caller:PrintMessage( 2 , "\n=======================" )
+    caller:PrintMessage( 2 , ply:Name() .. "'s friends:" )
+    caller:PrintMessage( 2 , "-----------------------" )
+    for target, status in pairs( friendTable ) do
+        caller:PrintMessage( 2 , target:Name() .. " : " .. status )
+    end
+    caller:PrintMessage( 2 , "=======================" )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:29]: May want to extract this out to a new helper function, seems out of place here.

File: lua/ulx/modules/sh/cfc_checkfriends.lua [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+    awaitingResponse[ply] = nil
+    ulx.fancyLogAdmin( caller, true, "#A checked #T's friends." , ply )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checkfriends.lua:32]: This has `#A` and `#T`, does it need more parameters after the `ply` ?

File: lua/ulx/modules/sh/cfc_checklua.lua [hunk 13]
```diff
@@ -0,0 +1,41 @@
+CFCUlxCommands.checklua = CFCUlxCommands.checklua or {}
+local cmd = CFCUlxCommands.checklua
+
+CATEGORY_NAME = "Utility"
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checklua.lua:4]:
```suggestion
local CATEGORY_NAME = "Utility"
```

File: lua/ulx/modules/sh/cfc_checklua.lua [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+    util.AddNetworkString( "CFC_ULX_StatCheckCL" )
+    util.AddNetworkString( "CFC_ULX_StatCheckSV" )
+end
+
+local awaitingResponse = {}
+
+net.Receive( "CFC_ULX_StatCheckSV", function( _, ply )
+    if awaitingResponse[ply] == nil then return end
+    local convar = net.ReadBool()
+    ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's sv_allowcslua value is " .. tostring( convar ), ply )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_checklua.lua:16]: You should be able to get away without using `tostring` here, as using string concatenation should do that for you.

```suggestion
    ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's sv_allowcslua value is " .. convar, ply )
```

File: lua/ulx/modules/sh/cfc_speed.lua [hunk 15]
```diff
@@ -0,0 +1,23 @@
+CFCUlxCommands.speed = CFCUlxCommands.speed or {}
+local cmd = CFCUlxCommands.speed
+
+CATEGORY_NAME = "Fun"
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_speed.lua:4]:
```suggestion
local CATEGORY_NAME = "Fun"
```

File: lua/ulx/modules/sh/cfc_speed.lua [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+                    
+function cmd.speedPlayers( callingPlayer, targetPlayers, amount )
+    for _, ply in pairs( targetPlayers ) do
+        local inPvp = hook.Run( "CFC_ULXCommands_CanApplySpeedModifier", ply )
+        --false = in pvp
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_speed.lua:9]:
```suggestion
        -- false = in pvp
```

File: lua/ulx/modules/sh/cfc_speed.lua [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+        if inPvp ~= false then
+            ply:SetWalkSpeed( 2 * amount )
+            ply:SetRunSpeed( 4 * amount )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_speed.lua:12]: Are these two values important? Should they be configurable be the user, or maybe at the top of the file / in a convar?

**Review by brandonsturgeon — CHANGES_REQUESTED** [review]
A few style concerns, but overall, LGTM!

File: lua/ulx/modules/sh/cfc_speed.lua (hunk 17 above)
Comment (wrefgtzweve) [reply lua/ulx/modules/sh/cfc_speed.lua:12]: They're the default movement speed values at 100% (100x2 and 100x4)

**Review by brandonsturgeon — APPROVED** [review]

**Comment by brandonsturgeon** [comment]
@wrefgtzweve Go ahead and make a changelog for this one

**Pull request merged**